lib/saml_idp/assertion_builder.rb

```diff
@@ -24,7 +24,6 @@ def initialize(
       name_id_format,
       x509_certificate,
       secret_key,
```

Comment:
is `secret_key` also unused?

Reply:
i think the answer to this is "no" but i think i want to do that in a separate PR, since i want to investigate all the uses of it

Reply:
makes sense!

Reply:
Confirmed, it is definitely in use in the IDP
- https://github.com/18F/identity-idp/blob/66a2b0cd78cc40eef989d5b551648291d819348c/app/controllers/concerns/saml_idp_auth_concern.rb#L215
- https://github.com/18F/identity-idp/blob/66a2b0cd78cc40eef989d5b551648291d819348c/app/services/saml_endpoint.rb#L41-L45

Reply:
yeah i saw that -- i'm not sure if it's necessary it be passed to so many classes within the saml_idp gem, but it does appear to be in use at some points.